include/openssl/pkcs7.h

```diff
@@ -105,27 +106,18 @@ OPENSSL_EXPORT int PKCS7_get_PEM_CRLs(STACK_OF(X509_CRL) *out_crls,
 //
 // These functions are a compatibility layer over a subset of OpenSSL's PKCS#7
 // API. It intentionally does not implement the whole thing, only the minimum
-// needed to build cryptography.io.
-
-typedef struct {
-  STACK_OF(X509) *cert;
-  STACK_OF(X509_CRL) *crl;
-} PKCS7_SIGNED;
-
-typedef struct {
-  STACK_OF(X509) *cert;
-  STACK_OF(X509_CRL) *crl;
-} PKCS7_SIGN_ENVELOPE;
-
-typedef void PKCS7_ENVELOPE;
-typedef void PKCS7_DIGEST;
-typedef void PKCS7_ENCRYPT;
-typedef void PKCS7_SIGNER_INFO;
-
-typedef struct {
-  uint8_t *ber_bytes;
-  size_t ber_len;
-
+// needed to build cryptography.io and CRuby.
+
+typedef struct pkcs7_st PKCS7;
+typedef struct pkcs7_signed_st PKCS7_SIGNED;
+typedef struct pkcs7_envelope_st PKCS7_ENVELOPE;
+typedef struct pkcs7_sign_envelope_st PKCS7_SIGN_ENVELOPE;
+typedef struct pkcs7_digest_st PKCS7_DIGEST;
+typedef struct pkcs7_encrypt_st PKCS7_ENCRYPT;
+typedef struct pkcs7_recip_info_st PKCS7_RECIP_INFO;
+typedef struct pkcs7_signer_info_st PKCS7_SIGNER_INFO;
```

Comment:
Nit: Should we keep this in `base.h` or do we like it here?

Reply:
yeah, that makes more sense.